tests/components/zha/test_sensor.py

```diff
@@ -300,6 +300,19 @@ async def async_test_device_temperature(hass: HomeAssistant, cluster, entity_id)
     assert_state(hass, entity_id, "29.0", UnitOfTemperature.CELSIUS)
 
 
+async def async_test_setpoint_change_source(hass, cluster, entity_id):
+    """Test the translation of numerical state into enum text."""
+    await send_attributes_report(hass, cluster, {0x0030: 0x01})
```

Comment:
`Thermostat.AttributeDefs.setpoint_change_source.id` instead of `0x0030`

Reply:
Done